src/univariate/discrete/binomial.jl

```diff
@@ -67,7 +67,7 @@ mean(d::Binomial) = ntrials(d) * succprob(d)
 var(d::Binomial) = ntrials(d) * succprob(d) * failprob(d)
 function mode(d::Binomial{T}) where T<:Real
     (n, p) = params(d)
-    n > 0 ? round(Int,(n + 1) * d.prob) : zero(T)
+    n > 0 ? round(Int,floor((n + 1) * d.p)) : zero(T)
```

Comment:
`rount(Int, floor(...))` can be written as `floor(Int, ...)`.

Reply:
Also it looks like this function is not type stable unless `T == Int`. It should either be `zero(Int)` instead of `zero(T)` or the `round`/`floor` call should be wrapped in `T()`.

Reply:
The latter is not a new problem and doesn't need to be fixed here.